### DISCUSSION TITLE
[mini.pairs] exiting spaced pairs

### DISCUSSION THREAD

**Discussion by TxHawks** [nvim-mini/mini.nvim#272] [Q&A] [answered]
I have the following mapping, which allows me to pad pairs (e.g., `(|)` + `space` -> `( | )` :

```lua
mappings = {
  [" "] = { 
    action = "closeopen",
    pair = "  ",
    neigh_pattern = "[%(%[{][%)%]}]",
  },
}
```

This works great and allows me to use another `space` to get to the closing `)` and then hit `)` to exit, which, generally, is usually what I want. BUT - the `space` to get to the `)` is both a wasted keystroke, and a bit confusing. What I'd really like is to be able to be in this position `( | )`, hit `)` and move to `(  )|`.

This will also allow me to convert the above rule from `closeopen` to just open.

Is this at all possible?

**Comment by echasnovski** [comment, marked as answer]
Unfortunately, no. Not with 'mini.pairs'. It is designed to know only about 1 character neighborhood and not further. You can try to make custom buffer mappings though, which will do as you suggest on appropriate wider neighborhood and utilize `MiniPairs.close()` otherwise.

**Reply by TxHawks** [reply]
Thanks. The current behavior is good-enough™
